tests/test_buffer.rs

```diff
@@ -1,3 +1,4 @@
+#![cfg(feature = "macros")]
```

Comment:
If several tests are deactivated and the others become harder to read, is this really better than requiring users to run
```
cargo test --features "auto-initialize macros"
```
or have the tests fail loudly otherwise?

Reply:
The end result of this PR is that `cargo test` will still run pretty much the whole pyO3 test suite, because the `macros` feature is enabled by default, and the change to `acquire_gil` means that pyo3's own tests effectively always have auto initialize enabled.

The reason a lot of test files had to get this cfg added is because `cargo test --no-default-features` now actually does disable the `macros` feature.

Reply:
> because the macros feature is enabled by default

Ah yes, this was the thing that I was missing.